ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureStorageCompactor.java

```diff
@@ -257,17 +309,61 @@ int purgeBlobs(List<CloudBlobMetadata> blobMetadataList) throws CloudStorageExce
   }
 
   /**
-   * Returns the dead blob in the specified partition with the earliest expiration or deletion time.
+   * Get the current compaction progress for a partition.
    * @param partitionPath the partition to check.
-   * @param fieldName the field name to use (expiration or deletion time).
-   * @return the {@link CloudBlobMetadata} for the dead blob, or NULL if none was found.
-   * @throws CloudStorageException
+   * @return a {@link Map} containing the progress time for compaction based on deletion and expiration time.
+   * @throws BlobStorageException if the checkpoint blob
+   */
+  Map<String, Long> getCompactionProgress(String partitionPath) throws BlobStorageException {
+    ByteArrayOutputStream baos = new ByteArrayOutputStream(64);
```

Comment:
why 64? Also maybe we can make it private static final

Reply:
approximate size of payload, made it static.